src/lib.rs

```diff
@@ -40,15 +40,51 @@ static mut GC_ALLOCATOR: GCMalloc = GCMalloc;
 
 static mut COLLECTOR: Option<Collector> = None;
 
+/// A garbage collected pointer. 'Gc' stands for 'Garbage collected'.
 #[derive(Debug)]
 pub struct Gc<T> {
     objptr: *mut T,
 }
 
+/// A garbage collected value is stored with a 1 machine word sized header. This
+/// header stores important metadata used by the GC during collection. It should
+/// never be accessible to users of the GC library. 
+///
+/// The metadata inside the header contains the mark-bit used by the collector
+/// to determine the value's reachability. It also contains the offset (in
+/// increments of usize) from the `objptr` to the base of the allocation block.
```

Comment:
What does "increments of usize" mean?

Reply:
I should reword this as "the offset from the `objptr to `baseptr` in machine words."

Reply:
Because this is unusual, I think you also need to add "in multiples of machine words" and then give an example (something like "e.g. if this has the value 8, then the base pointer is, on a 64-bit machine, 64 bytes before the object pointer"). [I think it's unusual for a good reason BTW!]

Do you have a check somewhere that this number always fits in the number of bits available to it? Just in case someone chucks us something aligned to (say) 128GiB or equally weird...

Reply:
> Do you have a check somewhere that this number always fits in the number of bits available to it? Just in case someone chucks us something aligned to (say) 128GiB or equally weird...

I don't. I make the assumption that it fits. This isn't very safe, but it might be worth just documenting it as such because this will be ripped out immediately in the next PR and replaced with a `vptr` when finalizers get put in.

Reply:
OK, in which case documenting it now is OK.

Reply:
Okay. See what you think of c484ee07086a94b97ecbe648796e987923e78155